[TensorExpr] LLVMCodegen: fix lowering for UInt->Float casts.

ZolotukhinM opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #64862

Previously we erroneously were looking at dst signedness. This was
discovered when we tried to implement quantize/dequantize ops.

Differential Revision: [D30881696](https://our.internmc.facebook.com/intern/diff/D30881696)

Commit:
[TensorExpr] LLVMCodegen: fix lowering for UInt->Float casts.

Previously we erroneously were looking at dst signedness. This was
discovered when we tried to implement quantize/dequantize ops.

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30881696).

navahgar commented on test/cpp/tensorexpr/test_llvm.cpp:
```diff
@@ -149,6 +149,34 @@ TEST(LLVM, ByteToDoubleCastTest) {
+  ASSERT_EQ(cg.value<int8_t>(), -2);
+}
+
+TEST(LLVM, BytoToFloatCastTest) {
```
typo: ByteToFloatCastTest

navahgar reviewed (approved):
LGTM

Commit:
Update on "[TensorExpr] LLVMCodegen: fix lowering for UInt->Float casts."


Previously we erroneously were looking at dst signedness. This was
discovered when we tried to implement quantize/dequantize ops.

Differential Revision: [D30881696](https://our.internmc.facebook.com/intern/diff/D30881696)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30881696).

bertmaher reviewed (approved):
I swear I’ve fixed multiple bugs here already.  I can’t believe more are falling out 🤦

ZolotukhinM commented:
> I swear I’ve fixed multiple bugs here already. I can’t believe more are falling out 🤦

Casts is a thing which we can exhaustively test sweeping through all their combos I think. If I find a nice way to do it, I'll send a PR.
